### PR TITLE
Add build template script + docs

### PR DESCRIPTION
**Pull request by weikanglim**
Build-Templates.ps1 for generating templates:

```pwsh
PS D:\repos\azure-dev> ./eng/scripts/Build-Templates.ps1 todo-java-mongo-aca
Gathering projects...
Found 1 project(s).
Generating todo-java-mongo-aca...
Successfully generated 1 project(s) in 00:00:07.2261082.
```

Documentation currently covers all local development and testing.

Does not include use of `test-templates.sh`, integration process , repository template staging process, `azd` dependency sequencing.

### CODE REVIEW

File: docs/develop-templates.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+```pwsh
+./eng/scripts/Build-Templates.ps1 python
+```
+
+Builds the template with names matching the regular expression `python`.
+
+## Build templates under a specific directory
+
+```pwsh
+./eng/scripts/Build-Templates.ps1 -Path ./templates/todo/projects/python-mongo
```
Comment (vhvb1989) [inline docs/develop-templates.md:47]: is it an option to integrate this logic directly to repoMan? (as we own that tool?)
Wondering about creating this pwsh file on top of the repoMan capabilities vs just adding the feature to repoMan...

The pwsh file script allow quick updates for the generation of templates, but it creates a dependency for repoMan ( we will need to be careful in the future about not introducing breaking changes between them)

Comment (weikanglim) [reply docs/develop-templates.md:47]: That's a good question. There are two features this script adds that could belong to `repoman`:
1. A command for auto-discovering nested `repo.yaml`(s) and calling generate on each
1. A flag for preserving a specific folder upon generation to preserve `.env` folder. 

Number 1. could be a desirable feature, although the use-case for having multiple nested templates is not common. The problem here is the integration with "automatic pushing to Git repository", where different templates might have different repository names. The way I think we could solve this is probably to break apart "folder/files stitching" from "repository syncing" as two separate tooling.

Number 2. is a potentially confusing feature outside of `repoman`. I think the generated directory should be 100% idempotent, but we want to preserve `.env` folder for local dev purposes.

Ultimately, whether we should add it to `repoman` should solely depend on whether it is a good feature for the tooling. Breaking changes are not really a good consideration -- if we make a breaking change to `repoman generate` or `repoman list`, obviously all tooling will break in some way. I'm inclined to say no for now.

File: docs/develop-templates.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    todo: # Templates related to the Simple ToDo application
+        api: # API servers definition
+        web: # Web servers definition
+        common: # Common assets shared across templates
+        projects: # The actual definition of how to assemble the template projects. repo.yml is defined here which references relevant assets from  different directories. 
+```
+
+## Building
+
+The simplest option is to build all templates in the repository:
```
[the remark this answers is not in the document]
Comment (weikanglim) [reply docs/develop-templates.md:18]: It runs the same logic, yes.

**Review by vhvb1989 — APPROVED** [review]

**Pull request merged**